code/ModelAdmin.php

```diff
@@ -487,6 +518,37 @@ public function getManagedModels()
         return $models;
     }
 
+    /**
+     * Get the tab name for a given model class
+     */
+    public function getTabForModelClass(string $modelClass): ?string
+    {
+        $managed = $this->getManagedModels();
+        if (is_subclass_of($modelClass, DataObject::class)) {
+            // Check the superclasses as well as the specifically passed-in class
+            $classes = array_reverse(ClassInfo::ancestry($modelClass));
+        } else {
+            // Most likely it was actually a tab name that was passed in, not a class name
+            $classes = [$modelClass];
```

Comment:
Not sure we should assume that something not subclassing DataObject should be a candidate here.

Reply:
There's a comment there that explains why this is being done - the tab _name_ can be explicitly defined. Allowing this method to check tab names as well as explicit classes means we can use it in the places it's being used in this PR.

The scenarios are:
1. A managed model _class_ is passed in. This method returns `true`.
2. A managed model _tab_ is passed in. This method returns `true`.
3. Literally anything other than a managed model or tab is passed in. This method returns `false`.

Reply:
> A managed model tab is passed in. This method returns the tab name.

What does it mean to pass in a `model tab` here?  As in the 'name' of the model tab? 

I'm not sure we should even have logic to support this here given the method name is `getTabForModelClass`

Reply:
If the reason to not support that is the name of the method, I'm happy to rename it to `getTabForModel` so it's not explicitly named 'class'. If the reason to not support it anything else then I'm happy to remove that part of the function of this method.

> What does it mean to pass in a `model tab` here? As in the 'name' of the model tab?

Yes. It's the `$this->modelTab`.

Reply:
Note that this is used for `isManagedModel()` which _does_ need to check tabs as well as classes, so if I remove that dual functionality form here I'll need to explicitly add it to that method.

Reply:
My first instinct was that the majority of the logic should've been in the `isManagedModel` anyway and then only have the `getTabForModel` to get the name, though I did understand that the way how the config for the tabs is constructed to allow for the title better supports the non-intuitive way. Perhaps we could have a private method normalising the config in a way we could use in both places?

A sidenote to Steve's comment — I would also ditch support for directly passing in a tab name as I'd say that's a non-standard way of defining which model classes the admin manages.

Reply:
Also minor thing, should change `getTabForModelClass` to `getModelTabForModelClass` to make things a little clearer